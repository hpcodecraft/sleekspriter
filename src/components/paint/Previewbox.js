import React from "react";
import PropTypes from "prop-types";
import { FrameCanvas } from "../canvases";

const Previewbox = props => {
  const maxWidth = 206,
    maxHeight = 120;

  let maxSize;
  if (props.size.width >= props.size.height) {
    maxSize = maxWidth; // scale to width
  } else {
    maxSize = maxHeight; // scale to height
  }

  let pixels;
  try {
    pixels = props.pixels[props.frame];
  } catch (e) {
    pixels = null;
  }

  return (
    <FrameCanvas
      frame={props.frame}
      layers={props.layers}
      size={props.size}
      maxSize={maxSize}
      noMargin={true}
      pixels={pixels}
    />
  );
};

Previewbox.propTypes = {
  size: PropTypes.object.isRequired,
  pixels: PropTypes.object.isRequired,
  layers: PropTypes.array.isRequired,
  frame: PropTypes.number.isRequired,
};

export default Previewbox;
