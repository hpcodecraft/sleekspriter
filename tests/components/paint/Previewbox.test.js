import Previewbox from "components/paint/Previewbox";

describe("Previewbox", () => {
  let props, jsx;
  beforeEach(() => {
    props = {
      size: { width: 32, height: 32 },
      pixels: {},
      layers: [],
      frame: 1,
      registerPreviewCanvas: jest.fn(),
    };
    jsx = <Previewbox {...props} />;
  });

  it("should render correctly", () => {
    expect(render(jsx)).toMatchSnapshot();
  });
});
