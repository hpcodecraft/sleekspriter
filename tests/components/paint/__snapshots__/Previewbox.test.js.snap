// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Previewbox should render correctly 1`] = `
<canvas
  height="192"
  width="192"
/>
`;
